### PR TITLE
feat:[HL-77] Registration API 🍕.

### PR DESCRIPTION
**Pull request by ahmednasr8898**
#### Proposed Changes

- api for register new account.

#### Related Issues

N/A.

#### Additional Information

N/A.

#### Screenshot

N/A.

#### Checklist

- [x] I have tested these changes locally.
- [ ] I have added appropriate documentation or updated existing documentation.
- [ ] I have added appropriate test coverage or updated existing test coverage.
- [ ] I have updated the changelog (if applicable).
- [x] I have followed the project's code style and formatting guidelines.
- [x] I have reviewed and adhered to the project's contributing guidelines.

### CODE REVIEW

File: Networking/Tests/NetworkingTests/Requests/RegisterRequestTests.swift [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+@testable import Networking
+
+final class RegisterRequestTests: XCTestCase {
+
+    
+    // MARK: - Lifecycle
+    
+    override func setUp() {
+        
+    }
```
Comment (ahmdmhasn) [inline Networking/Tests/NetworkingTests/Requests/RegisterRequestTests.swift]:
```suggestion
```

File: Networking/Sources/Networking/Requests/RegisterRequest.swift [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import Foundation
+
+// MARK: - RegisterResponse
+public struct RegisterResponse: Codable {
+    let idToken, email, refreshToken, expiresIn: String
+    let localID: String
+
+    enum CodingKeys: String, CodingKey {
+        case idToken, email, refreshToken, expiresIn
```
Comment (ahmdmhasn) [inline Networking/Sources/Networking/Requests/RegisterRequest.swift]: - idToken needed to be rewritten
- You need to convert expires in to date

File: Networking/Sources/Networking/Requests/RegisterRequest.swift [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import Foundation
+
+// MARK: - RegisterResponse
+public struct RegisterResponse: Codable {
+    let idToken, email, refreshToken, expiresIn: String
+    let localID: String
```
Comment (ahmdmhasn) [inline Networking/Sources/Networking/Requests/RegisterRequest.swift]:
```suggestion
    public let idToken, email, refreshToken, expiresIn: String
    public let localID: String
```

**Review by ahmdmhasn — APPROVED** [review]

File: Networking/Sources/Networking/Requests/RegisterRequest.swift [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+// MARK: - RegisterRequest
+public struct RegisterRequest: RequestType {
+    public typealias ResponseType = RegisterResponse
+
+    private let email: String
+    private let password: String
+    private let returnSecureToken: Bool
+    
+    public init(email: String, password: String, returnSecureToken: Bool = true) {
```
Comment (ahmdmhasn) [inline Networking/Sources/Networking/Requests/RegisterRequest.swift:23]: Thanks for injecting the parameters 🙏🏼

**Review by djayyab — APPROVED** [review]

**Review by oradwanomar — APPROVED** [review]

**Pull request merged**